src/berryimu.py

```diff
@@ -0,0 +1,243 @@
+from typing import ClassVar, Mapping, Tuple
+from typing import ClassVar, Mapping, Any, Dict, Optional
+from typing_extensions import Self
+
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import ResourceName
+from viam.resource.base import ResourceBase
+from viam.resource.types import Model, ModelFamily
+
+from viam.components.movement_sensor import MovementSensor
+from viam.logging import getLogger
+from viam.proto.common import GeoPoint, Orientation, Vector3
+from smbus import SMBus
+from . import constants
+from . import utils
+import time
+import math
+import threading
+import asyncio
+from .orientation_vector import euler_angles_to_orientation_vector
+
+
+LOGGER = getLogger(__name__)
+dt = 0.08
+
+class Berryimu(MovementSensor):
+    MODEL: ClassVar[Model] = Model(ModelFamily("viam", "movement_sensor"), "berryimu")
+    i2cbus: SMBus
+    acceleration: Vector3
+    accelerometer_raw: Vector3
+    velocity: Vector3
+    orientation: Orientation
+    compass_heading: float
+    accelerometer_angles: Vector3
+    gyro_angles: Vector3
+    euler_angles: Vector3
+    calibrate: bool
+    
+
+    @classmethod
+    def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
+        berryimu = cls(config.name)
+        berryimu.reconfigure(config, dependencies)
+        b = threading.Thread(name='readings', target=berryimu.read)
+        b.start()
```

Comment:
we might want to check the success of your start method, have you seen any instances where it does not start correctly? (i2c pins not connected for example)

Reply:
will do some more testing

Reply:
in the case that the i2c pins are not connected correctly reconfigure will fail and this thread will never start, eventually you get `configuration error: reached max of 5 configuration attempts for rdk:component:movement_sensor/imu `. What should the expected behavior be in this case?

Reply:
That's expected behavior - add the readme that the modules expect the pins to be connected at startup.